src/main/java/com/transferwise/openbanking/client/api/registration/RestRegistrationClient.java

```diff
@@ -116,15 +122,29 @@ public ClientRegistrationResponse updateRegistration(ClientRegistrationRequest c
         }
     }
 
-    private String getClientCredentialsToken(AspspDetails aspspDetails) {
-        // The spec states a scope value isn't strictly needed, but some ASPSPs do actually require it, additionally
-        // some do not accept the general `openid` scope and require either `accounts` or `payments` scope.
-        // We could take the value from the `SoftwareStatementDetails.permissions` property, but if the TPP is trying to
-        // update the scope of their registration, this will contain a permission that we can't use here. So rather than
-        // trying to figure out what the TPP has vs what they are requesting, we use `payments` as given this library
-        // only has payments support it's most likely the TPP currently has this permission on their registration.
-        GetAccessTokenRequest getAccessTokenRequest = GetAccessTokenRequest.clientCredentialsRequest("payments");
+    private String getClientCredentialsToken(AspspDetails aspspDetails,
+                                             SoftwareStatementDetails softwareStatementDetails) {
+        String scope = generateScopeValue(aspspDetails, softwareStatementDetails);
+        GetAccessTokenRequest getAccessTokenRequest = GetAccessTokenRequest.clientCredentialsRequest(scope);
         AccessTokenResponse accessTokenResponse = oAuthClient.getAccessToken(getAccessTokenRequest, aspspDetails);
         return accessTokenResponse.getAccessToken();
     }
+
+    private String generateScopeValue(AspspDetails aspspDetails, SoftwareStatementDetails softwareStatementDetails) {
+        // The spec states a scope value isn't strictly needed, but some ASPSPs do actually require it, additionally
+        // some require the scope to contain `openid` but some require it to not contain `openid`. As we request a
+        // scope of what the permissions the software statement details currently has, we don't really support updating
+        // the permissions of a client registration, but as this can't be modified in the Open Banking directory this
+        // shouldn't be an issue.
+        Set<RegistrationPermission> permissions = new LinkedHashSet<>(softwareStatementDetails.getPermissions());
```

Comment:
Using a linked hash set here so that we get consistent order of elements, to make the tests simpler. Not using a list here instead because we still want the unique element constraint to handle both cases of the software statement permissions including openid and not including openid. (As a side note, we might want to refactor this later so the permissions can only be payments and/or accounts)